### PR TITLE
Describe validation errors from createShaderModule

### PR DESCRIPTION
**Pull request by kainino0x**
changes without whitespace:
https://github.com/gpuweb/gpuweb/pull/3347/files?w=1

issue: #2308

### CODE REVIEW

**Review by toji — APPROVED** [review]
LGTM

File: spec/index.bs [hunk 1]
```diff
@@ -5869,33 +5873,40 @@ dictionary GPUShaderModuleDescriptor : GPUObjectDescriptorBase {
             <div data-timeline=device>
                 [=Device timeline=] |initialization steps|:
 
-                1. If any of the following conditions are unsatisfied
-                    [$generate a validation error$], make |sm| [=invalid=], and stop.
+                1. Let |result| be the result [=shader module creation=] with the WGSL source
```
Comment (dneto0) [inline spec/index.bs:5876]: missing "of"? 

>  be the result of [=shader module creation=]....

File: spec/index.bs [hunk 2]
```diff
@@ -5869,33 +5873,40 @@ dictionary GPUShaderModuleDescriptor : GPUObjectDescriptorBase {
...
+                    |descriptor|.{{GPUShaderModuleDescriptor/code}}.
+                1. If any of the following requirements are unmet,
+                    [$generate a validation error$], make |sm| [=invalid=], and return.
 
                     <div class=validusage>
-                        - |this| is [=valid=].
+                        - |this| must be [=valid=].
+                        - |result| must not be a [=program error=].
```
Comment (dneto0) [inline spec/index.bs:5883]: Better to use [=shader-creation error=] here

https://gpuweb.github.io/gpuweb/wgsl/#shader-creation-error 

It's the set of errors that can (and should) be detected at createShaderModule time.


At lot of the "must" statements in the WGSL spec are links to [=shader-creation-error=]

**Review by dneto0 — CHANGES_REQUESTED** [review]

File: spec/index.bs (hunk 2 above)
Comment (kainino0x) [reply spec/index.bs:5883]: I wanted to avoid capturing "uncategorized errors" here. Does "shader-creation error" exclude those?

Comment (kainino0x) [reply spec/index.bs]: @dneto0 PTAL

**Review by dneto0 — APPROVED** [review]
Thanks for your patience!

File: spec/index.bs [hunk 3]
```diff
@@ -5869,33 +5873,40 @@ dictionary GPUShaderModuleDescriptor : GPUObjectDescriptorBase {
...
-                    [$generate a validation error$], make |sm| [=invalid=], and stop.
+                1. Let |result| be the result of [=shader module creation=] with the WGSL source
+                    |descriptor|.{{GPUShaderModuleDescriptor/code}}.
+                1. If any of the following requirements are unmet,
+                    [$generate a validation error$], make |sm| [=invalid=], and return.
 
                     <div class=validusage>
-                        - |this| is [=valid=].
+                        - |this| must be [=valid=].
+                        - |result| must not be a [=program error|program=] [=shader-creation error=].
```
Comment (kainino0x) [inline spec/index.bs:5883]:
```suggestion
                        - |result| must not be a [=shader-creation=] [=program error=].
```

File: spec/index.bs [hunk 4]
```diff
@@ -5869,33 +5873,40 @@ dictionary GPUShaderModuleDescriptor : GPUObjectDescriptorBase {
...
-                    [$generate a validation error$], make |sm| [=invalid=], and stop.
+                1. Let |result| be the result of [=shader module creation=] with the WGSL source
+                    |descriptor|.{{GPUShaderModuleDescriptor/code}}.
+                1. If any of the following requirements are unmet,
+                    [$generate a validation error$], make |sm| [=invalid=], and return.
 
                     <div class=validusage>
-                        - |this| is [=valid=].
+                        - |this| must be [=valid=].
+                        - |result| must not be a [=shader-creation=] [=program error=].
```
Comment (kainino0x) [inline spec/index.bs:5883]:
```suggestion
                        - |result| must not be a [=shader-creation error|shader-creation=] [=program error=].
```

**Pull request merged**